vernac/debugHook.ml

```diff
@@ -89,3 +88,12 @@ module Intf = struct
   let get () = !ltac_debug_ref
 
 end
+
+let debug_proof = ref None (* only need to save sigma and goals here *)
+let proof_data () =
+  let p = Proof.data (Vernacstate.Declare.give_me_the_proof ()) in
+  let open Proof in
+  match !debug_proof with
+  | Some (sigma, goals) -> { p with sigma; goals }
+  | None -> p
+  [@@ocaml.warning "-3"];;
```

Comment:
Drops `sigma` and `goals` into `Proof.t`.  Don't know how to get the updated `stack`.